packages/plugin-ext/src/main/node/plugin-deployer-impl.ts

```diff
@@ -132,69 +137,75 @@ export class PluginDeployerImpl implements PluginDeployer {
         }
     }
 
-    async deploy(pluginEntry: string, type: PluginType = PluginType.System): Promise<void> {
+    async deploy(plugin: UnresolvedPluginEntry): Promise<void> {
         const startDeployTime = performance.now();
-        await this.deployMultipleEntries([pluginEntry], type);
+        await this.deployMultipleEntries([plugin]);
         this.logMeasurement('Deploy plugin entry', startDeployTime);
     }
 
-    protected async deployMultipleEntries(pluginEntries: ReadonlyArray<string>, type: PluginType = PluginType.System): Promise<void> {
```

Comment:
Is this (and `resolvePlugins`) the change that requires the introduction of the  `UnresolvedPluginEntry` interface? It looks like previously we could only have a list of plugins with uniform plugin types, and now we want to have a list of plugin with mixed types. Otherwise, it looks like the same information was being passed around everywhere, just not necessarily in the same package?

Reply:
Exactly, we need the mixed type since we need to resolve **all** user and builtin extensions in the first step. Though it's mostly `resolvePlugins`, since the main culprit of this issue is introduced here:

https://github.com/eclipse-theia/theia/blob/c56aa230fe50f20b598b2b4c4273cad4486b4750/packages/plugin-ext/src/main/node/plugin-deployer-impl.ts#L121-L124